bin/commander.js

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const { readFileSync, writeFileSync, existsSync, lstatSync } = require('fs');
+const { execSync } = require('child_process');
+const program = require('commander');
+
+const VERSION_REGEX = /\* Version:(.+)/;
+const VERSION_CONSTANT_REGEX = /define\(\s*'WEBSTORIES_VERSION',\s*'([^']*)'\s*\);/;
+const STABLE_TAG_REGEX = /Stable tag:\s*(.+)/;
+
+const PLUGIN_DIR = process.cwd();
+const README_FILE = PLUGIN_DIR + '/readme.txt';
+const PLUGIN_FILE = PLUGIN_DIR + '/web-stories.php';
+const DISTIGNORE_FILE = PLUGIN_DIR + '/.distignore';
+
+/**
+ * Updates version numbers in plugin files.
+ *
+ * Namely, this updates the 'Version' header in the main plugin file,
+ * the `WEBSTORIES_VERSION` constant in the same file, as well as the
+ * stable tag in the readme.txt.
+ *
+ * If a specific version is provided, uses that version to update the constant.
+ *
+ * The stable tag is only updated if it's not a pre-release.
+ *
+ * @param {string} version Desired version number.
+ * @param {boolean} isPrerelease Whether this is a pre-release.
+ * @return {void} Nothing.
+ */
+function updateVersionNumbers(version = undefined, isPrerelease = false) {
+  // Get the current commit hash. Used for pre-releases.
+  const currentCommitHash = execSync('git rev-parse HEAD')
+    .toString()
+    .trim()
+    .slice(0, 7);
+
+  let pluginFileContent = readFileSync(PLUGIN_FILE, 'utf8');
+  const currentVersion = pluginFileContent.match(VERSION_REGEX)[1].trim();
+
+  isPrerelease = isPrerelease || currentVersion.includes('-');
+
+  if (version) {
+    // 'Version' plugin header must not include anything else beyond the version number,
+    // i.e. no suffixes or similar.
+    pluginFileContent = pluginFileContent.replace(
+      VERSION_REGEX,
+      `* Version: ${version}`
+    );
+  }
+
+  const newVersion =
+    version ||
+    (isPrerelease ? `${currentVersion}-${currentCommitHash}` : currentVersion);
```

Comment:
For [semver](https://semver.org/) it should probably be like this:

```suggestion
    (isPrerelease ? `${currentVersion}+${currentCommitHash}` : currentVersion);
```